packages/babel-types/src/definitions/core.js

```diff
@@ -704,7 +704,7 @@ defineType("WithStatement", {
       object: assertNodeType("Expression")
     },
     body: {
-      validate: assertNodeType("BlockStatement")
+      validate: assertNodeType("BlockStatement", "Statement")
```

Comment:
Because Statement is a superset?
I should do the same for the `WhileStatement` and other places this is happening